### ISSUE TITLE
(ec2): add support for IMDSv1 metadata in L2 construct

### ISSUE THREAD

**Issue by NGL321** [aws/aws-cdk#19080]
### Description

Currently the L1 construct [`CfnLaunchTemplate`](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-ec2.CfnLaunchTemplate.html) for ec2 instances allows for defining metadata resources in its [`LaunchTemplateDataProperty`](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-ec2.CfnLaunchTemplate.LaunchTemplateDataProperty.html). The L2 [`LaunchTemplate`](https://github.com/aws/aws-cdk/blob/v2.13.0/packages/@aws-cdk/aws-ec2/lib/launch-template.ts#L576-L580) always provides a fixed property in the creation of the L1 resource.

### Use Case

Greater metadata flexibility within the creation of EC2 instances as well as explicitly disabling IMDSv1.

### Proposed Solution

Add optional construct prop with simplification of L1 `LaunchTemplateDataProperty`.

### Other information

_Request from EC2 team Ticket-D24130265_

### Acknowledge

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by jungseoklee** [comment]
> explicitly disabling IMDSv1.

Doesn't it require breaking/backward-incompatible change?

**Comment by NGL321** [comment]
@jungseoklee I dont think it necessitates a breaking change. So long as the new property is optional with a default override of the current behavior, it shouldnt introduce any breaking changes
